pkg/helper/client.go

```diff
@@ -242,6 +246,63 @@ func ApplyDesiredState(desiredState nmstatev1alpha1.State) (string, error) {
 	return commandOutput, nil
 }
 
+// There is a bug in Kernel/NetworkManager on systems with NetworkManager
+// 1.20, where sometimes after disconnecting a NIC from a bonding, the NIC
+// remains in 'unavailable' state and cannot be used for a new connection. This
+// is likely caused by an issue with autonegotiation where the NIC appears to
+// be disconnected and the only thing that can bring it available again is
+// explicitly calling `ip link set <name> up` on it. In order to workaround
+// this issue until it gets solved, we iterate all devices during `nmstatectl
+// set` and if we find some with 'unavailable' we explicitly set them up.
+func setUnavailableUp(stopCh chan struct{}) {
```

Comment:
Let's implement it with https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L486 so we don't re-invento the wheel.